### ISSUE TITLE
click the app  in the dock cannot keep the language setting

### ISSUE THREAD

**Issue by ljhui** [micheleg/dash-to-dock#736]
I set chrome language to chinese in the desktop file, in the gnome overview, click the icon, I can get a chinese chrome, but when I add it to favourites in the dock , click the icon in the dock, what I get is always a english chrome. if I disable the dash to dock extension, click chrome app on the system left dock, it can get a chinese chrome. 
chrome version: 66.0.3359.139
system version: 4.16.5-200.fc27.x86_64
gnome version: 3.26.2
how to fix it?

**Comment by micheleg** [comment]
Hi,

can you copy here the relevant part of the .desktop file? Could you also do this check. Turn of the extension, and use the default system left dock. You say in this case the language settings is respected. If instead of left clicking the launcher you right click on it and select the "New window" option, does the program respect the language?

**Comment by ljhui** [comment]
system dock with the "New window" option doesn't respect the language, but 
thanks for your reminder, I have solved it. the chrome default desktop file have three part,`[Desktop Entry]`,`[Desktop Action NewWindow]` and `[Desktop Action NewPrivateWindow]` , I only modify the `Exec` to `sh -c "LANG=zh_CN.utf-8   /usr/bin/google-chrome-stable  %U"` in the`[Desktop Entry]`  , if I also modify the  `Exec` item in `[Desktop Action NewWindow]`,click app icon or right click and select the "New window" option works well.

**Comment by Znert** [comment]
Hi,

I just installed Manjaro and got the same issue, but with Firefox.

I need to set the theme for FF to light one, otherwise some input fields become too dark. So I added `env GTK_THEM=light_theme` to each of the Execs:

    Exec=env GTK_THEME=Adapta-Eta-Maia /usr/lib/firefox/firefox %u

However, I do not get the theme to work as it should. Starting it from the application menu or the dashboard seems to work just fine.

Is this some caching issue of an old .desktop file? Restarting the PC and/or adding/removing FF from my favourites does not work either though.
